src/components/DetailsList/DetailsList.tsx

```diff
@@ -224,7 +225,7 @@ export class DetailsList extends React.Component<IDetailsListProps, IDetailsList
         data-is-scrollable='false'
         aria-label={ ariaLabel }
         role={ shouldApplyApplicationRole ? 'application' : '' }>
-        <div role='grid' aria-label={ ariaLabelForGrid }>
+        <div role={ shouldApplyGridRole ? 'grid' : '' } aria-label={ ariaLabelForGrid }>
```

Comment:
Well... do you think it would be better if I do this? 
role= { this.props.ariaRoleForGrid }